### ISSUE TITLE
Cookies

### ISSUE THREAD

**Issue by andrewfairlie** [madebymutual/GDPR-ePR-Guide#1]
Cookie polices are really up in the air at the moment.

GDPR wants opt-in for any data collection, and I think that they really want opt-in for different usage types. EG: Site cookies separate from analytics cookies.

And then ePR is suggesting that the responsibility of cookie notices moves from site devs to the browser manufacturers.

Anyone any clearer about a safe and legal way to comply with GDPR cookies if ePR isn't ready in time - or if they remove that section from ePR in the final version?

**Comment by Tam** [comment]
I found this a while back, so it might not apply to GDPR (it's from 2012) but it does expand a bit on what cookies are excempt from concent:

http://ec.europa.eu/ipg/basics/legal/cookies/index_en.htm#section_2

> However, some cookies are exempt from this requirement. Consent is not required if the cookie is:
>
>   - used for the sole purpose of carrying out the transmission of a communication, and
>   - strictly necessary in order for the provider of an information society service explicitly required by the user to provide that service.
>
> Cookies clearly exempt from consent according to the [EU advisory body on data protection- WP29](http://ec.europa.eu/justice/data-protection/article-29/documentation/opinion-recommendation/files/2012/wp194_en.pdf) include:
>
>   - **user‑input** cookies (session-id) such as first‑party cookies to keep track of the user's input when filling online forms, shopping carts, etc., for the duration of a session or persistent cookies limited to a few hours in some cases
>   - **authentication** cookies, to identify the user once he has logged in, for the duration of a session
>   - **user‑centric security** cookies, used to detect authentication abuses, for a limited persistent duration
>   - **multimedia content player** cookies, used to store technical data to play back video or audio content, for the duration of a session
>   - **load‑balancing** cookies, for the duration of session
>   - **user‑interface customisation** cookies such as language or font preferences, for the duration of a session (or slightly longer)
>   - **third‑party social plug‑in content‑sharing** cookies, for logged‑in members of a social network.

TLDR: Everything is fine, as long as it's not analytics 😆

**Comment by andrewfairlie** [comment]
The ePrivacy Directive is (probably) being replaced by the ePrivacy Legislation which changes the rules.

I think the directive is being pulled because countries implemented it inconsistently, and it's not fully compatible with GDPR.

But these rules are sensible, so they hopefully stand.

What about Analytics then? 😬 😢

**Comment by Tam** [comment]
The only mention of cookies in the entire document is this (highlighting added):

> (30) Natural persons may be associated with online identifiers provided by their devices, 
applications, tools and protocols, such as internet protocol addresses, **cookie identifiers or 
other identifiers** such as radio frequency identification tags. This may leave traces which, 
in particular when combined with unique identifiers and other information received by the 
servers, may be **used to create profiles of the natural persons and identify them**.

Which reads (at least as far as we understand it internally) as meaning you are not allowed to store any data which may be used to build up a profile of the user AND be linked back to that specific individual (be it via user ID, email, or any other piece of personal information). 

(The reference to cookies is... poorly worded. Does it mean that cookies are inherently identifiers, or simply a method of storing the identifier? I'm pretty sure who ever wrote this only mentioned cookies because their grandson knows that "it tracks you".)

This leads us to believe that anonymized analytical tracking is permissable because there is no way of linking the analytical profile with a specific individual.

Reading around that section (I just did a find for "cookie") there is also this, which confirms our suspicions about anonymized data (again, highlighting added):

> (26) The principles of data protection should apply to any information concerning an identified 
or identifiable natural person. Personal data which have undergone pseudonymisation, 
which could be attributed to a natural person by the use of additional information should be 
considered to be information on an identifiable natural person. To determine whether a 
natural person is identifiable, account should be taken of all the means reasonably likely to 
be used, such as singling out, either by the controller or by another person to identify the 
natural person directly or indirectly. To ascertain whether means are reasonably likely to 
be used to identify the natural person, account should be taken of all objective factors, such 
as the costs of and the amount of time required for identification, taking into consideration 
the available technology at the time of the processing and technological developments. **The 
principles of data protection should therefore not apply to anonymous information, namely 
information which does not relate to an identified or identifiable natural person or to 
personal data rendered anonymous in such a manner that the data subject is not or no 
longer identifiable. This Regulation does not therefore concern the processing of such 
anonymous information, including for statistical or research purposes.**

So for analytics, at least for the time being, you DON'T need to ask the users permission.

However, GDPR is not specific to the web (far from it, I'm pretty sure the first draft was written on a typewriter). When ePR comes into affect it will probably turn all of this on it's head and say "You can't track anyone, even anonymously, unless the user fills out a concent form, signs it in triplicate, and posts copies to everyone that has ever had anything to do with the website and the government of every member state."

Basically, the smartest move is to become a hermit.

**Comment by Tam** [comment]
Also, https://blog.littledata.io/2017/10/19/is-google-analytics-compliant-with-gdpr/

**Comment by croxton** [comment]
I agree with this interpretation:
> This leads us to believe that anonymized analytical tracking is permissable because there is no way of linking the analytical profile with a specific individual.

This is what we've been telling clients that use Google Analytics:

> Currently you only use fully-anonymised data collection via Google Analytics to track users of your website. Since it is impossible to relate the data collected to any individual, we believe that you will not require prior consent to the use of cookies via a cookie pop-up or other UI widget.
> 
> However, you are required to list and describe the purpose of the cookies that you use on your website on a cookie page. You should review your use of cookies on the website, and make sure this page is up-to-date.
> 
> In future should you decide to use tracking software that is able to identify individuals then you must first put in place a cookie consent mechanism that keeps a record of consents granted. In which case we would advise integrating a webservice such as Cookiebot https://www.cookiebot.com.

**Comment by andrewfairlie** [comment]
Yeah, makes sense. Thanks for input @Tam @croxton

**Comment by paul-frost** [comment]
I've just seen this Craft plugin that covers Google analytics as part of the consent request:
https://github.com/zauberware/acceptcookies

But I think we need an official answer to the specific question of "Does Google Analytics need consent", rather than all of us independently trying to guess or read between the lines an interpretation that we think we might get away with. Given the world wide use of GA I would think X million websites need to know the official answer, and there shouldn't be any caveats that it may or may not depending etc... It's either yes or no.
If there are variables, based on the exact code used and if it has the IP address etc. then publish examples of the GA code snippet that could be safely used without consent or warning required.

**Comment by croxton** [comment]
I don't think you'll get an 'official' answer since it's a matter of legal opinion and there is no practical guidance or case law to refer to. We work for a few well-known law firms who all use GA, and so far only one has decided that they _must_ use a cookie consent widget (Cookiebot, which was deemed to be GDPR compliant because it keeps a record of consents and allows consent to be withdrawn). If the lawyers aren't sure then I'm not sure how we devs can say definitively one way or another. Best to let your client make the decision I think.

**Comment by paul-frost** [comment]
I guess I'm surprised that what seems like a black/white situation, it either does cross the line or it doesn't, is so vague that even lawyers can't decide themselves.
It might be OK for big firms to make a calculated judgement and be prepared to defend it in court, but I manage sites for small sports clubs and they are looking to me to provide guidance as they don't understand what happens behind the scenes with cookies and tracking.
Could we be held responsible if we haven't advised our clients that their site should be asking for consent?

**Comment by croxton** [comment]
You shouldn't, no, unless you have misrepresented your recommendations to your client as constituting legal advice.

Make sure in any communications with your client about GDPR that you have a disclaimer that makes it clear that you not offering legal advice, and recommend that they seek legal advice.

Hopefully you have a contract with them already that contains a data protection clause? That will protect you in the event of a data breach.

Finally, if you're worried you can buy cyber / data risk insurance (usually as part of professional indemnity insurance), and/or set yourself up as a limited company to limit your liability should you be sued.

**Comment by paul-frost** [comment]
Thanks Mark, but I'm not making a living out of building websites for clubs. In some cases I build and manage them as a volunteer. Others I charge a nominal amount to cover my costs. I think of myself as retired now and only do this because I enjoy it and to help improve the image of the sports. There are a lot of very old and amateurish sites out there.  So setting myself up as a limited company isn't going to happen.

I will probably end up just removing all the GA tracking code and Facebook and Twitter widgets from all my sites. The analytics is useful to monitor visitor trends and devices and highlight where navigation could be improved, but it's not the end of the world to loose it. The facebook and twitter feeds will be missed by some users and will probably lead to less visitors to the website. But making them have to keep seeing cookie warnings will probably piss them off anyway and drive them away.

They will happily give facebook and twitter permission to track them but will view other websites asking as suspicious.
